### ISSUE TITLE
[Bug] WithClaims should bypass token cache

### ISSUE THREAD

**Issue by bgavrilMS** [AzureAD/microsoft-authentication-library-for-dotnet#1639]
**Which Version of MSAL are you using ?**
Note that to get help, you need to run the latest version. Preview version are also ok. 
For ADAL, please log issues to https://github.com/AzureAD/azure-activedirectory-library-for-dotnet
4.8.1

**Platform**
all

**What authentication flow has the issue?**
AquireTokenSilent

**Repro**

```csharp
               await app
                    .Silent(TestConstants.s_scope)
                    .WithClaims(TestConstants.Claims)```
```

**Expected behavior**
An AT token from the cache MUST not be returned. The refresh_token grant must be performed so that an AT with those claims is returned.

Cache is bypassed by ADAL and by MSAL for Android 

**Actual behavior**
Cache is not bypassed

Part of the work for Client Capabilities https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1545

**Comment by jmprieur** [comment]
wow @bgavrilMS : is the claims a key of the cache? how does it work today: we should see infinite loops?

**Comment by bgavrilMS** [comment]
claims is not a key of the cache. Today if you pass Claims to AcquierTokenSilent, it would return you an access token that is not tied to those claims. So Claims only works with `AcquireTokenInteractive` really.

**Comment by jmprieur** [comment]
Oh, yes that makes sense. today we get a claims challenge and call AcquireTokenInteractive.
Good point for client capabilities;

**Comment by trwalke** [comment]
released in 4.9.0
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.9.0
